### DISCUSSION TITLE
Anyone wants connection limit user per IP?

### DISCUSSION THREAD

**Discussion by amir-devman** [XTLS/Xray-core#2271] [Ideas]
Anyone is interested in adding ability to have connection limit by IP?
Meaning that each user while having at least one session of established TCP connection from an IP no one else can have successful connection.
I can go research and implement it on xray-core but I'm a newbie to golang.

Just wondering if anyone is interested?

**Comment by AbadanVpn** [comment]
It is a very good idea and I hope you can do it

**Reply by khkhhasan** [reply]
Hello, good time, if possible, please add the limit of the number of devices to xray core instead of the IP limit, if both are at the same time, that will be great.

**Comment by iSegaro** [comment]
We definitely need this feature
Please do this

**Comment by mahdi383ir** [comment]
Its very good, nice👌

**Comment by meaqese** [comment]
In normal VPN services, this will not make sense, because they usually have several servers. Commerce is certainly good, but let the service owners deal with it themselves, no extra code is needed.

**Comment by Nima786** [comment]
It's a must have feature which unfortunately is missing...

**Comment by w0l4i** [comment]
We definitely need this feature
Please do this

**Comment by w0l4i** [comment]
@hiddify1

**Comment by TheyCallMeSecond** [comment]
There is no need,xray panels has this function already

**Reply by w0l4i** [reply]
🤔 not in core bruh we mean every uuid could be managing to how much ip(session per user ) can be used at the same time

**Reply by TheyCallMeSecond** [reply]
better to add this function to panel and keep the core lightweight

**Reply by w0l4i** [reply]
> better to add this function to panel and keep the core lightweight

c'mon , this could be a plug in to core 🥲

**Reply by w0l4i** [reply]
> better to add this function to panel and keep the core lightweight

also there are panels that have this function built-in but they have a lot of issues , if this option will be added so much problem could be gone !

**Reply by TheyCallMeSecond** [reply]
Do you sell vpn bro? 😂 What problem?

**Reply by w0l4i** [reply]
> Do you sell vpn bro? 😂 What problem?

the problem is if you don't need the feature, it doesn't matter what others need ! , this could be an extension in the core .
end of descution

**Comment by Mohammad12261** [comment]
We definitely need this feature
Please do this

**Comment by Mohammad12261** [comment]
We definitely need this feature
Please do this

**Comment by ach1992** [comment]
We definitely need this feature
Please do this

**Comment by SajjadSedaqat** [comment]
It would be great

**Comment by w0l4i** [comment]
@safasafari @hiddify1 @vahidfarid we need your help to make it !

**Reply by hiddify-com** [reply]
we have prepared some stuffs but we need more time for testing

**Comment by Jaylennyy** [comment]
It's handy. Thanks so much

**Comment by dremaratkar** [comment]
We definitely need this feature
Please do this

**Comment by xlirezx** [comment]
It'll be really helpful 👍🏻👌🏻

**Reply by xlirezx** [reply]
Unless it makes the core heavyweight for cheap VDSs

**Reply by w0l4i** [reply]
> Unless it makes the core heavyweight for cheap VDSs

this can be an extension who wants to make it on who doesn't default is off 🤔

**Comment by MrCjxer** [comment]
Yessss
We all want it

**Comment by amir-devman** [comment]
Thanks everyone for having positive opinion for this discussion, here's the beta PR: https://github.com/XTLS/Xray-core/pull/2277

**Comment by khkhhasan** [comment]
Hello, good time, if possible, please add the limit of the number of devices to xray core instead of the IP limit, if both are at the same time, that will be great.

**Comment by loskiq** [comment]
We definitely need this feature
Please do this
